Source/Core/Common/Config/ConfigInfo.h

```diff
@@ -30,10 +34,30 @@ struct Location
 };
 
 template <typename T>
-struct Info
+struct CachedValue
 {
-  Info(const Location& location_, const T& default_value_)
-      : location{location_}, default_value{default_value_}
+  T value;
+  u64 config_version;
+};
+
+template <typename T>
+class Info
+{
+public:
+  constexpr Info(const Location& location, const T& default_value)
+      : m_location{location}, m_default_value{default_value}, m_cached_value{default_value, 0}
+  {
+  }
+
+  Info(const Info<T>& other)
+      : m_location{other.GetLocation()}, m_default_value{other.GetDefaultValue()},
+        m_cached_value{other.GetCachedValue()}
+  {
+  }
+
+  Info(Info<T>&& other)
+      : m_location{std::move(other.m_location)}, m_default_value{std::move(other.m_default_value)},
+        m_cached_value{std::move(other.m_cached_value)}
```

Comment:
Why is it safe to not lock the mutex here?

Reply:
It's not that it's safe to do this without locking the mutex, it's that moving the object while another thread wants to use it doesn't make any sense even if we lock the mutex. The other thread will either read a valid setting or read some useless data depending on timing, and I can't think of any scenario where that would be acceptable.